packages/server/test/integration/cypress_spec.js

```diff
@@ -349,6 +350,48 @@ describe('lib/cypress', () => {
       sinon.stub(commitInfo, 'getRemoteOrigin').resolves('remoteOrigin')
     })
 
+    describe('cloud recommendation message', () => {
+      it('gets logged when in CI and there is a failure', function () {
+        process.env.CI = true
+        delete process.env.CYPRESS_NO_COMMERCIAL_RECOMMENDATIONS
+        globalThis.CY_TEST_MOCK.listenForProjectEnd = { stats: { failures: 1 } }
+
+        return cypress.start([`--run-project=${this.todosPath}`]).then(() => {
+          expect(console.log).to.be.calledWith(cloudRecommendationMessage)
+        })
+      })
+
+      it('does not get logged if CYPRESS_NO_COMMERCIAL_RECOMMENDATIONS is set', function () {
+        process.env.CI = true
+        process.env.CYPRESS_NO_COMMERCIAL_RECOMMENDATIONS = true
+        globalThis.CY_TEST_MOCK.listenForProjectEnd = { stats: { failures: 1 } }
+
+        return cypress.start([`--run-project=${this.todosPath}`]).then(() => {
+          expect(console.log).not.to.be.calledWith(cloudRecommendationMessage)
+        })
+      })
+
+      it('does not get logged if all tests pass', function () {
+        process.env.CI = true
+        delete process.env.CYPRESS_NO_COMMERCIAL_RECOMMENDATIONS
+        globalThis.CY_TEST_MOCK.listenForProjectEnd = { stats: { failures: 0 } }
+
+        return cypress.start([`--run-project=${this.todosPath}`]).then(() => {
+          expect(console.log).not.to.be.calledWith(cloudRecommendationMessage)
+        })
+      })
+
+      it('does not get logged if not running in CI', function () {
+        delete process.env.CI
+        delete process.env.CYPRESS_NO_COMMERCIAL_RECOMMENDATIONS
+        globalThis.CY_TEST_MOCK.listenForProjectEnd = { stats: { failures: 1 } }
```

Comment:
Nice job finding a way to test this, unit testing `run.js` is often tricky.

Reply:
yeah, it was tricky. one thing that I found (and I still don't understand why this happens) is that I couldn't get my test case to work like this

```js
it('does something', () => {
  // my test
})
```
and I had to declare the callback function like this

```js
it('does something', function () {
  // my test
})
```

it took me a while to figure out that my test was failing because I was trying to define the callback as an arrow function 🤔 still don't know why